p11kit/attribute.go

```diff
@@ -78,6 +78,25 @@ func (o *Object) SetLabel(label string) {
 	})
 }
 
+// SetCertificate set certificate attributes to key object.
+// It will create the link between key and certificate.
+func (o *Object) SetCertificate(cert *x509.Certificate) error {
+	keyType, exist := o.attributeValue(attributeClass)
+	if !exist {
+		return fmt.Errorf("CKA_CLASS doesn't exist")
+	}
+	if keyType.ulong == nil || (*keyType.ulong != ckoPrivateKey && *keyType.ulong != ckoPublicKey) {
+		return fmt.Errorf("SetCertificate supports only PublicKey and PrivateKey object")
+	}
+
+	attrs, err := extractCertAttriForKey(cert)
```

Comment:
nit: For readability, maybe avoid the function call since you're not using it elsewhere. E.g. just compute the attributes in this function.